src/control/cmd/daos/container.go

```diff
@@ -1297,20 +1297,24 @@ func (cmd *containerSetPropCmd) Execute(args []string) error {
 	}
 	defer deallocCmdArgs()
 
-	ap.props = cmd.Args.Props.props
-
 	cleanup, err := cmd.resolveAndConnect(C.DAOS_COO_RW, ap)
 	if err != nil {
 		return err
 	}
 	defer cleanup()
 
-	rc := C.cont_set_prop_hdlr(ap)
+	rc := C.daos_cont_set_prop(ap.cont, cmd.Args.Props.props, nil)
 	if err := daosError(rc); err != nil {
-		return errors.Errorf("failed to set properties on container %s",
+		return errors.Wrapf(err, "failed to set properties on container %s",
 			cmd.ContainerID())
 	}
 
+	if cmd.jsonOutputEnabled() {
+		return cmd.outputJSON(nil, nil)
+	}
```

Comment:
I don't _think_ this is necessary... Have you tried it without? The way it should work is that if `--json` was supplied, the `cmd.Info` output is suppressed, so the end result should be the same as if you call `cmd.outputJSON(nil, nil)`. If it's not, that's a bug IMO.